File: Smallest subarray with sum greater than x - GFG/smallest-subarray-with-sum-greater-than-x.java
//{ Driver Code Starts
//Initial Template for Java

import java.util.*;
import java.lang.*;
import java.io.*;

class GFG {
	public static void main(String[] args) throws IOException
	{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(br.readLine().trim());
        while(t-->0)
        {
            StringTokenizer stt = new StringTokenizer(br.readLine());
            
            int n = Integer.parseInt(stt.nextToken());
            int m = Integer.parseInt(stt.nextToken());
            // int n = Integer.parseInt(br.readLine().trim());
            int a[] = new int[n];
            String inputLine[] = br.readLine().trim().split(" ");
            for (int i = 0; i < n; i++) {
                a[i] = Integer.parseInt(inputLine[i]);
            }
            
            Solution obj = new Solution();
            System.out.println(obj.smallestSubWithSum(a, n, m));
        }
	}
}


// } Driver Code Ends


//User function Template for Java


class Solution {

    public static int smallestSubWithSum(int arr[], int n, int x) {
        // Your code goes here 
     int i=0;
     int j=0;
     int sum=0;
     
     int ans=Integer.MAX_VALUE;
     while(j<n)
     {
        sum+=arr[j];
        if(sum>x)
        {
            ans=Math.min(ans,j-i+1);
        
         while(sum > x)
           {
            // ans=Math.min(ans, j-i+1);
            sum-=arr[i];
            i++;
            if(sum>x)
            ans=Math.min(ans,j-i+1);
           }
         }
        j++;
     }
     if(ans==Integer.MAX_VALUE)
     ans=0;
     return ans;
    }
}

File: Smallest subarray with sum greater than x - GFG/smallest-subarray-with-sum-greater-than-x.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;


// } Driver Code Ends
class Solution{
  public:

    int smallestSubWithSum(int arr[], int n, int x)
    {
        // Your code goes here
         int len = -1, sum = 0, j = 0;
    for(int i=0; i<n;) {
        if(arr[i] > x)
            return 1;
        sum += arr[i];
        if(sum > x) {
            if(len == -1 || len > i - j + 1)
                len = i - j + 1;
            sum -= arr[j++];
            sum -= arr[i];
        }
        else {
            i++;
        }
    }
    return len == -1 ? 0 : len;
    }
};

//{ Driver Code Starts.

int main() {
	// your code goes here
	int t;
	cin>>t;
	while(t--)
	{
		int n,x;
		cin>>n>>x;
		int a[n];
		for(int i=0;i<n;i++)
		cin>>a[i];
		Solution obj;
		cout<<obj.smallestSubWithSum(a,n,x)<<endl;
	}
	return 0;
}
// } Driver Code Ends